src/components/lib/cos_kernel_api.c

```diff
@@ -787,7 +787,7 @@ cos_tcap_transfer(arcvcap_t dst, tcap_t src, tcap_res_t res, tcap_prio_t prio)
 int
 cos_tcap_delegate(asndcap_t dst, tcap_t src, tcap_res_t res, tcap_prio_t prio, tcap_deleg_flags_t flags)
 {
-	u32_t yield     = flags & TCAP_DELEG_YIELD;
+	u32_t yield     = (flags && TCAP_DELEG_YIELD);
```

Comment:
I'm not sure I can understand why this would be the desirable code.

It is functionally equivalent to `u32_t yield = (flags != 0);`.  If the intention is to see if the correct flag (i.e. bit) in the mask is set, then that is what `&` is made for.

If the previous code really is the problem, then I don't think that you understand the problem.  This feels like a fix for the symptoms, not a fix for the problem.